model/histogram/float_histogram.go

```diff
@@ -307,13 +307,16 @@ func (h *FloatHistogram) Sub(other *FloatHistogram) *FloatHistogram {
 // Exact match is when there are no new buckets (even empty) and no missing buckets,
 // and all the bucket values match. Spans can have different empty length spans in between,
 // but they must represent the same bucket layout to match.
+// Non-metadata float fields (Sum, Count, ZeroCount) are compared as binary (using math.Float64bits).
```

Comment:
"Metadata" is a highly overloaded word.

Suggestion: "Sum, Count, and ZeroCount are compared based on their bit patterns because this method is about data equality rather than mathematical equality."